### ISSUE TITLE
Implement XGBoost Multiclass

### ISSUE THREAD

**Issue by Aylr** [HealthCatalyst/healthcareai-py#282]
@Aylr commented on [Sat Jun 03 2017](https://github.com/HealthCatalyst/healthcareai-r/issues/465)

## Implementation

- [ ] algorithm in AdvancedTrainer
- [ ] tests (and possibly a new dataset?)
- [ ] wrapper in Trainer
- [ ] check data prep pipeline for implications (think about binary converter - that needs to change) - Should this be another pipeline? Or logic in this pipeline?
- [ ] metrics: confusion matrix (array and plot)
- [ ] need new example_multi_classes_1.py (and possibly a new dataset to showcase the example?)

## Things to note

- There will be metrics ramifications that need to be thought through
- Sklearn's random forest can do multi class - consider simultaneous implementation? (If so, create a new issue)

**Comment by Aylr** [comment]
Why was this closed?

**Comment by mxlei01** [comment]
Do you mean multi-label?

**Comment by Aylr** [comment]
yes

**Comment by Aylr** [comment]
Also, I've started and have a good chunk of this done.
